### PR TITLE
planner: not to generate partial agg when cop task has root conditions

### LINKED ISSUE #15110: TiDB failed in compile time when isolation read set to tiflash
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
CREATE TABLE `crm_rd_150m` (
`product` varchar(256) DEFAULT NULL,
`uks` varchar(16) DEFAULT NULL,
`brand` varchar(256) DEFAULT NULL,
`cin` varchar(16) DEFAULT NULL,
`created_date` timestamp NULL DEFAULT NULL,
`quantity` int(11) DEFAULT NULL,
`amount` decimal(11,0) DEFAULT NULL,
`pl_date` timestamp NULL DEFAULT NULL,
`customer_first_date` timestamp NULL DEFAULT NULL,
`recent_date` timestamp NULL DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin;

alter table crm_rd_150m set tiflash replica 1;

set tidb_isolation_read_engines="tiflash"

mysql> explain SELECT count(*) FROM `crm_rd_150m` `dataset_48` WHERE (CASE WHEN (month(`dataset_48`.`customer_first_date`)) <= 30 THEN '新客' ELSE NULL END) IS NOT NULL;
ERROR 1105 (HY000): Can't find column test.crm_rd_150m.customer_first_date in schema Column: Column#14 Unique key: []

2. What did you expect to see?
the execution plan without error

3. What did you see instead?
ERROR 1105 (HY000): Can't find column test.crm_rd_150m.customer_first_date in schema Column: Column#14 Unique key: []

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
tidb master

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve? 
fix #15110 

### What is changed and how it works?
not to generate partial aggregation when cop task has root conditions

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch (release 3.1)

### CODE REVIEW

File: planner/core/task.go [hunk 1]
```diff
@@ -1169,7 +1169,11 @@ func (p *PhysicalStreamAgg) attach2Task(tasks ...task) task {
 		// We should not push agg down across double read, since the data of second read is ordered by handle instead of index.
 		// The `extraHandleCol` is added if the double read needs to keep order. So we just use it to decided
 		// whether the following plan is double read with order reserved.
-		if cop.extraHandleCol == nil {
+		if cop.extraHandleCol != nil || len(cop.rootTaskConds) > 0 {
```
Comment (francis0407) [inline planner/core/task.go]: Only `Aggregation` is affected? How about `TopN` and `Limit`?

Comment (lzmhhh123) [reply planner/core/task.go]: Yes, you are right.

**Pull request merged**